app/server/src/routes/space/space.controller.js

```diff
@@ -165,6 +167,42 @@ const SpaceController = {
       res.status(400).send({ error: e.toString() });
     }
   },
+  getRecommendedSpaces: async function (req, res) {
+    try {
+      const user_id = req.auth.id;
+      const user = await UserModel.User.findById(user_id);
+      const personalInfo = await PersonalInfoModel.PersonalInfo.findOne({_id: user.personal_info});
+      var interests = personalInfo.interests;
+      const url = "https://api.datamuse.com/words?";
+      var inferred_interests = [];
+      for (interest_t of interests) {
+        if (interest_t.includes(" ")) interest_t = interest_t.replace(" ", "+");
+        const trg_url = `${url}max=5&ml=${interest_t}`;
+        const trg_result = await axios.get(trg_url);
+        inferred_interests = inferred_interests.concat(trg_result.data);
+      }
+      inferred_interests.sort((a, b) => b.score - a.score);
+      var spaces = [];
+      var space_ids = [];
+      for (inf_in of inferred_interests) {
+        interests.push(inf_in.word);
+      }
+      for (interest_t of interests) {
+        let spaces_t = await SpaceModel.Space.find({$text: {$search: `\"${interest_t}\"`}})
+                              .limit(2)
+                              .exec();
+        for (space_t of spaces_t) {
+          if (!space_ids.includes(space_t._id.toString())) {
```

Comment:
burda bir de daha önce enroll olduğu spaceleri exclude etmek için check yapalım

Reply:
done!